open_rarity/resolver/opensea_api_helpers.py

```diff
@@ -308,25 +349,42 @@ def get_collection_from_opensea(
     # Fetch token metadata
     tokens: list[Token] = []
     batch_size = 30
-    num_batches = math.ceil(total_supply / batch_size)
     initial_token_id = 0
 
-    # Returns a list of `batch_size` token IDs, such that no token ID
-    # can exceed `max_token_id` (in which case len(return_value) < `batch_size`)
-    def get_token_ids(
-        batch_id: int, max_token_id: int = total_supply - 1
-    ) -> list[int]:
-        token_id_start = initial_token_id + (batch_id * batch_size)
-        token_id_end = int(min(token_id_start + batch_size - 1, max_token_id))
-        return list(range(token_id_start, token_id_end + 1))
-
-    for batch_id in range(num_batches):
-        token_ids = get_token_ids(batch_id)
-        tokens_batch = get_tokens_from_opensea(
-            opensea_slug=opensea_collection_slug, token_ids=token_ids
+    # Return list of lists for batched token_ids
+    def batch_token_ids(
+        intial_token_id: int = 0,
+        max_token_id: int = total_supply - 1,
+        batch_size: int = 30,
+    ):
+        token_ids = list(range(initial_token_id, max_token_id))
+        return array_split(
+            token_ids,
+            len(token_ids) // batch_size + 1,
```

Comment:
hm shouldn't this be `math.ceil(len(token_ids / batch_size))`?

If we have 10 tokens of batch size 3, we want 4 groups - 3, 3, 3, 1 => your code does this properly
If we have 12 tokens of batch size 3, we want 4 groups - 3, 3, 3, 3 => your code only does 3 batches instead of 4